### PR TITLE
Support 0-dimensional shapes as inputs/outputs to view()

### LINKED ISSUE #2383: python view (reshape) op doesn't support zero dim tensors
```
RuntimeError: !original_sizes.empty() INTERNAL ASSERT FAILED at "../torch/csrc/jit/codegen/cuda/ops/alias.cpp":87, please report a bug to PyTorch. No support for 0-dim tensors in view support.
```

### PR DESCRIPTION
**Pull request by jacobhinkle**
Fixes #2383

### CODE REVIEW

File: third_party/nvfuser/test/test_gpu_view.cpp [hunk 1]
```diff
@@ -2285,5 +2285,118 @@ TEST_F(NVFuserTest, FusionIssue2076_v2_CUDA) {
...
+  FusionExecutor fe;
+  fe.compileFusion(&fusion, aten_inputs, lparams);
+  auto outputs = fe.runFusion(aten_inputs, lparams);
+
+  auto at_prod = (at_x.squeeze() + at_y.squeeze()) * at_z;
+
+  testValidate(&fusion, outputs, aten_inputs, {at_prod}, __LINE__, __FILE__);
+}
+
+TEST_F(NVFuserTest, FusionViewZeroDimInputOutput_CUDA) {
```
Comment (kevinstephano) [inline third_party/nvfuser/test/test_gpu_view.cpp]: Change `FusionView...` to `FusionReshape...`

File: third_party/nvfuser/test/test_gpu_view.cpp [hunk 2]
```diff
@@ -2285,5 +2285,118 @@ TEST_F(NVFuserTest, FusionIssue2076_v2_CUDA) {
...
+  FusionGuard fg(&fusion);
+
+  TensorView* x = makeSymbolicTensor({}, DataType::Float);
+  fusion.addInput(x);
+  TensorView* y = makeSymbolicTensor({}, DataType::Float);
+  fusion.addInput(y);
+
+  auto x_view = view(x, {}, {});
+  // test broadcasting then squeezing
+  x_view = view(x, {}, {1, 1, 1});
```
Comment (kevinstephano) [inline third_party/nvfuser/test/test_gpu_view.cpp]: `view` -> `reshape`

File: third_party/nvfuser/test/test_gpu_view.cpp [hunk 3]
```diff
@@ -2285,5 +2285,118 @@ TEST_F(NVFuserTest, FusionIssue2076_v2_CUDA) {
   testValidate(&fusion, cg_outputs, {t0, t1, t2}, {t4, t6}, __LINE__, __FILE__);
 }
 
+TEST_F(NVFuserTest, FusionViewZeroDimInput_CUDA) {
```
Comment (kevinstephano) [inline third_party/nvfuser/test/test_gpu_view.cpp]: `FusionView...` -> `FusionReshape...`

File: third_party/nvfuser/test/test_gpu_view.cpp [hunk 4]
```diff
@@ -2285,5 +2285,118 @@ TEST_F(NVFuserTest, FusionIssue2076_v2_CUDA) {
...
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  TensorView* x = makeSymbolicTensor({}, DataType::Float);
+  fusion.addInput(x);
+  TensorView* y = makeSymbolicTensor(3, DataType::Float);
+  fusion.addInput(y);
+
+  auto x_view = view(x, {}, {1, 1, 1});
```
Comment (kevinstephano) [inline third_party/nvfuser/test/test_gpu_view.cpp]: `view` -> `reshape`

**Comment by kevinstephano** [comment]
I just changed `view` operations to be named `reshape` operations that could cosmetically impact your code.  In particular, the tests.  The context is that our `view` ops were creating new tensors and Pytorch's `view` is an inplace operation.  `reshape` creates a new tensor.  So, to alleviate confusion, we did a rename.

**Comment by jacobhinkle** [comment]
I just saw your PR was merged. Currently renamed everything just running tests then will push with the changes.

**Review by kevinstephano — APPROVED** [review]
LGTM

Note, going from [1, 1, 1] -> scalar tensor already worked.

**Pull request merged**